package.json

```diff
@@ -5,6 +5,7 @@
   "version": "4.0.9",
   "author": "Stanford University Office of Development",
   "dependencies": {
+    "@emotion/react": "^11.11.1",
```

Comment:
@wmui51,

This is a new CSS convention for this website and it requires a [large bundle](https://bundlephobia.com/package/@emotion/react@11.11.1). Is it possible to avoid adding this package and using what exists today or are you blocked by something?

Reply:
I have thoughts but perhaps @yvonnetangsu and/or @rebeccahongsf have better thoughts first?

Reply:
@wmui51 @sherakama I haven't looked into this PR, but could we use `dcnb` (for dynamically chaining class names)?

Reply:
@wmui51 Hmm Thanks for the preview - that really helps. I'm wondering if this UX works well for this situation - seems a bit tough to make it work well on mobile also. Personally I feel like the donut chart design is a bit difficult to understand (especially for "days"). I wonder if there's chance to go back to the team and revisit the design before we decide on the dev approach?

cc @sherakama @rebeccahongsf

Reply:
Would rather us not have to install emotion just for this one thing 😄

Reply:
Thanks, all, for your thoughts on how best to implement this countdown timer. A similar timer will (probably) be included in the emails going out to donors using this service: https://countdownmail.com/ The team wanted to visually match the look of the countdown timer, hence the "rings" around the numbers. All that being said, I am happy to go back to the team and propose removing the rings to make the implementation cleaner and more efficient.

Reply:
Ok, I just spoke with the project lead, and she has a strong desire to keep the rings as part of the display. She feels it creates an added urgency to take action before the time has run out and provides a similar visual to what users have seen in the email driving traffic to the page. As a fallback to writing our own version of the timer, we could also embed a third-party countdown timer – assuming we could find one that meets our needs.

Reply:
@forrestglick @wmui51 Ahh I see - totally understand trying to match the look of the email timer. As someone who hasn't seen the email, my initial impression was that the rings add to the cognitive load, and I was more drawn to thinking about the ring fragments instead of on the time left instead (which probably isn't what you all want 😄 ). Would be awesome if we could change the design - if not, I think css vars might be worth an exploration @wmui51 👍🏼

Reply:
@forrestglick Sorry I replied before seeing your latest message 😬

Reply:
@yvonnetangsu Yes, I agree the rings add to the cognitive load, and I also want to defer to the project lead and designer. Do know that I've explained a bit about the complexities of the implementation, so they are at least aware of what is involved. I think it would be great if Wilbert explored alternatives to loading the emotion package. If we use it for this one page, would that JS package get loaded on any call to the site? Any means of keeping the code clean/efficient sounds good to me!

Reply:
@forrestglick Thanks for bringing that up 😄 I believe we could try different approach before installing the package. @wmui51 I've had success in the past by creating an empty `div` to replace a pseudo element. Perhaps give that a try?

Reply:
@wmui51 Yes, I was thinking of a service that basically loaded as an iframe-type embed. That way, it would only load when that specific page was rendered. I leave the explorations to you all, as I'm clearly out of my depth.

Reply:
I'm fine with iframe embed as well if we could find one that matches the look and feel, provided that it's accessible 😄 @forrestglick @wmui51

Reply:
I brief search online surfaced this option: https://logwork.com/countdown-1or7

Reply:
That looks really promising @forrestglick 😄  ! Any chance the text (days, hours etc) can be a bit bigger. They're quite tiny on mobile.

Reply:
@yvonnetangsu I do not see a way to control for text size – only colors can be adjusted.

Reply:
@wmui51 and @yvonnetangsu, 

This is a SCSS and JavaScript based product. Why not just programmatically generate all the styles using a loop and rotate through the class names?

Reply:
@sherakama That works too - we were using a loop to generate TW-liked classes for background colors etc in this repo already, so open to using a loop t generate rotation classes as well. Though I'm ok with limited use of inline styles as well - since we do use that with dynamic background images and transforms.

Reply:
Thanks @wmui51 

I appreciate the refactor. I am fine with this approach.

Reply:
@wmui51 Also fine with this approach - thank you 👍🏼